pkg/kubernetes/kubernetes.go

```diff
@@ -421,24 +433,30 @@ func getCRDFromManifest(ctx context.Context, fileName string) (*apiextensionsv1b
 // return error if failure observed
 func GetDynamicInformer(ctx context.Context, crdGroup string, crdVersion string, crdName string, namespace string) (informers.GenericInformer, error) {
 	log := logger.GetLogger(ctx)
-	// Get a config to talk to the apiserver
-	cfg, err := GetKubeConfig(ctx)
-	if err != nil {
-		log.Errorf("failed to get Kubernetes config. Err: %+v", err)
-		return nil, err
-	}
-	// Grab a dynamic interface to create informers from
-	dc, err := dynamic.NewForConfig(cfg)
-	if err != nil {
-		log.Errorf("could not generate dynamic client for config. err :%v", err)
-		return nil, err
+	dynamicInformerInitLock.Lock()
+	defer dynamicInformerInitLock.Unlock()
+
+	if !dynamicInformerInitialized {
+		// Get a config to talk to the apiserver
+		cfg, err := GetKubeConfig(ctx)
+		if err != nil {
+			log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+			return nil, err
+		}
+		// Grab a dynamic interface to create informers from
+		dc, err := dynamic.NewForConfig(cfg)
+		if err != nil {
+			log.Errorf("could not generate dynamic client for config. err :%v", err)
+			return nil, err
+		}
+		if namespace != "" {
+			dynamicInformerFactory = dynamicinformer.NewFilteredDynamicSharedInformerFactory(dc, 0, namespace, nil)
```

Comment:
What is the use case to support dynamic informer factory on one namespace? I see an issue with this code where the first `GetDynamicInformer` callers passes ns-1 and the next caller passes empty string, both of them using the informer factory initialized only for ns-1. 

Since we are creating this as a singleton, we should create the factory instance for all namespaces by default.

Reply:
Currently there are no callers that specify namespace so i'll remove the option to provide a namespace and default to all namespaces.